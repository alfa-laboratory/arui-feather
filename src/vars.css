/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

@import './vars/font.css';
@import './vars/color.css';
@import './vars/color-theme_alfa-on-color.css';
@import './vars/color-theme_alfa-on-white.css';
@import './vars/opacity.css';
@import './vars/border-radius.css';
@import './vars/shadow.css';

:root {
    --small-screen-padding: 16px;
    --small-header-height: 80px;
    --large-screen-content-width: 1000px;
    --xlarge-screen-content-width: 1100px;
}
