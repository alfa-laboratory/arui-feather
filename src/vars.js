/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

/* eslint max-len: 0 */

export const SCROLL_TO_CORRECTION = 16;
export const SCROLL_TO_DEFAULT_DURATION = 0;
export const SCROLL_TO_NORMAL_DURATION = 250;
export const SCROLL_TO_EASING = 'easeInOutSine';

export const POPUP_MAIN_OFFSET = 12;

export const LOREM_IPSUM = [
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. ',
    'Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum. ',
    'Donec nec justo eget felis facilisis fermentum. Aliquam porttitor mauris sit amet orci. Aenean dignissim pellentesque felis. ',
    'Morbi in sem quis dui placerat ornare. Pellentesque odio nisi, euismod in, pharetra a, ultricies in, diam. Sed arcu. Cras consequat. ',
    'Praesent dapibus, neque id cursus faucibus, tortor neque egestas augue, eu vulputate magna eros eu erat. Aliquam erat volutpat. Nam dui mi, tincidunt quis, accumsan porttitor, facilisis luctus, metus. ',
    'Sed adipiscing ornare risus. Morbi est est, blandit sit amet, sagittis vel, euismod vel, velit. Pellentesque egestas sem. Suspendisse commodo ullamcorper magna. ',
    'Etiam maximus velit quis massa lacinia laoreet. Curabitur egestas, massa vel ullamcorper cursus, sem augue faucibus orci, eu mattis erat mi at arcu. In at ligula suscipit, bibendum nisl vel, hendrerit ex. ',
    'Quisque sagittis vehicula augue a tristique. Suspendisse quis accumsan quam, placerat ornare sapien. Phasellus consectetur nibh non est molestie malesuada. ',
    'Nullam tortor ex, commodo id tristique non, auctor a mi. Donec at feugiat dui, in efficitur ex. Fusce molestie tortor quis leo mattis lobortis. Maecenas luctus risus lorem, id aliquet eros commodo at. ',
    'Fusce lacus sapien, euismod gravida purus non, dignissim sollicitudin arcu. Ut facilisis massa eros, vel hendrerit eros luctus at. Sed dictum ligula ex, et viverra mauris varius non. ',
    'Sed rhoncus condimentum velit at fringilla. Mauris dignissim posuere vestibulum. Nulla sed velit vel ligula semper varius. Quisque gravida nec massa eu fringilla. ',
    'Duis risus ligula, iaculis vel rutrum eget, varius vitae massa. Proin non accumsan purus. Fusce non augue in enim ullamcorper aliquet a quis quam. ',
];
