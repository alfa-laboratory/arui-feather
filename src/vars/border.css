/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

:root {
    /**
     * Стандартное скругление углов.
     */
    --border-radius-control: 4px;

    /**
     * Стиль рамки на неактивных полях ввода.
     */
    --border-style-control-disabled: dashed;
}
