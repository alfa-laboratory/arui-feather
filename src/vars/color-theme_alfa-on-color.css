/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

:root {
    /**
     * Цвет контента с повышенным акцентом для темы `alfa-on-color`.
     * Используется, чтобы компенсировать потерю акцента для контента.
     *
     * Например для заголовков страниц, набранных тонким шрифтом, что делает
     * их "слишком светлыми" при использовании стандартного `--color-content-alfa-on-color`.
     */
    --color-content-accent-alfa-on-color: rgb(255, 255, 255);

    /**
     * Цвет контента для темы `alfa-on-color`.
     */
    --color-content-alfa-on-color: rgba(255, 255, 255, .9);

    /**
     * Цвет второстепенного контента для `alfa-on-color`.
     * Например, используется на неактивных полях ввода.
     */
    --color-content-minor-alfa-on-color: rgba(255, 255, 255, .3);

    /**
     * Цвет фона компонентов для `alfa-on-color`.
     */
    --color-background-alfa-on-color: rgba(0, 0, 0, 1);

    /**
     * Цвет фона для контролов для `alfa-on-color`.
     */
    --color-background-control-alfa-on-color: transparent;

    /**
     * Цвет фона для контролов при наведении для `alfa-on-color`.
     */
    --color-background-control-hovered-alfa-on-color: rgba(255, 255, 255, .1);

    /**
     * Цвет фона для контролов в фокусе для `alfa-on-color`.
     */
    --color-background-control-focused-alfa-on-color: rgba(0, 0, 0, .15);

    /**
     * Цвет фона для замороженных контролов для `alfa-on-color`.
     */
    --color-background-control-disabled-alfa-on-color: rgba(255, 255, 255, .1);

    /**
     * Цвет рамки для контролов в фокусе для `alfa-on-color`.
     */
    --color-border-control-focused-alfa-on-color: rgba(255, 255, 255, .9);
}
