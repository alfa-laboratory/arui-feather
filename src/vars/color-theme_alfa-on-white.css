/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

:root {
    /**
     * Цвет контента с особым акцентом для темы `alfa-on-white`.
     */
    --color-content-heavy-accent-alfa-on-white: rgba(178, 12, 0, 1);

    /**
     * Цвет контента с повышенным акцентом для темы `alfa-on-white`.
     * Используется, чтобы компенсировать потерю акцента для контента.
     *
     * Например для заголовков страниц, набранных тонким шрифтом, что делает
     * их "слишком светлыми" при использовании стандартного `--color-content-alfa-on-white`.
     */
    --color-content-accent-alfa-on-white: rgb(11, 31, 53);

    /**
     * Цвет контента для темы `alfa-on-white`.
     */
    --color-content-alfa-on-white: rgba(11, 31, 53, .95);

    /**
     * Цвет второстепенного контента для `alfa-on-white`.
     */
    --color-content-minor-alfa-on-white: rgba(11, 31, 53, .6);

    /**
     * Цвет неактивного контента для `alfa-on-white`.
     * Например, используется на неактивных полях ввода.
     */
    --color-content-disabled-alfa-on-white: rgba(11, 31, 53, .3);

    /**
     * Цвет фона компонентов для `alfa-on-white`.
     */
    --color-background-alfa-on-white: rgb(255, 255, 255);

    /**
     * Цвет фона для для второстепенного контента `alfa-on-white`.
     */
    --color-background-minor-alfa-on-white: rgba(11, 31, 53, .08);

    /**
     * Цвет фона для контролов для `alfa-on-white`.
     */
    --color-background-control-alfa-on-white: rgba(255, 255, 255, .9);

    /**
     * Цвет фона для контролов при наведении для `alfa-on-white`.
     */
    --color-background-control-hovered-alfa-on-white: rgba(11, 31, 53, .08);

    /**
     * Цвет фона для контролов в фокусе для `alfa-on-white`.
     */
    --color-background-control-focused-alfa-on-white: rgba(11, 31, 53, .08);

    /**
     * Цвет фона для замороженных контролов для `alfa-on-white`.
     */
    --color-background-control-disabled-alfa-on-white: rgba(11, 31, 53, .08);

    /**
    * Цвет рамки для контролов для `alfa-on-white`.
    */
    --color-border-control-alfa-on-white: rgba(11, 31, 53, .4);

    /**
     * Цвет рамки для контролов в фокусе для `alfa-on-white`.
     */
    --color-border-control-focused-alfa-on-white: rgba(11, 31, 53, .9);

    /**
     * Цвет рамки для контролов при наведении для `alfa-on-color`.
     */
    --color-border-control-hovered-alfa-on-white: rgba(11, 31, 53, .9);

    /**
     * Цвет рамки для нажатых контролов для `alfa-on-color`.
     */
    --color-border-control-pressed-alfa-on-white: rgba(11, 31, 53, .9);

    /**
    * Цвет рамки для выбранных контролов с фоном для `alfa-on-color`.
    */
    --color-border-control-checked-alfa-on-white: rgba(11, 31, 53, .5);

    /**
     * Цвет рамки для неактивных контролов для `alfa-on-color`.
     */
    --color-border-control-disabled-alfa-on-white: rgba(11, 31, 53, .2);

    /**
    Цвет для разделительных бордеров для `alfa-on-white`.
    */
    --color-border-separator-alfa-on-white: rgba(11, 31, 53, .07);

    /**
    * Цвет для градиента на краях экрана для `alfa-on-white`.
    */
    --color-gradient-edge-screen-alfa-on-white: linear-gradient(to top, #fff, rgba(255, 255, 255, 0));
}
