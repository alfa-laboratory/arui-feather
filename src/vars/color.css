/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

:root {
    /**
     * Основной акцентный цвет.
     */
    --color-accent: rgba(240, 50, 38, 1);

    /**
     * Основной акцентный цвет с небольшой прозрачностью. Может быть использован для
     * второстепенных акцентных элементов. Например, используется для бордеров
     * акцентных кнопок.
     */
    --color-accent-translucent: rgba(240, 50, 38, .85);

    /**
     * Основной акцентный цвет с сильной прозрачностью. Может быть использован для
     * второстепенных акцентных элементов в неактивном состоянии.
     * Например, используется для неактивного состояния акцентных кнопок.
     */
    --color-accent-heavy-translucent: rgba(240, 50, 38, .25);

    /**
     * Дополнительный к `--color-accent` акцентный цвет. Может быть использован
     * для отображения состояния компонентов, использующих `--color-accent`.
     * Например, состояние "зажатое" для акцентной кнопки.
     */
    --color-accent-alt: rgba(158, 12, 0, 1);

    /**
     * Цвет контента на акцентном фоне.
     */
    --color-accent-content: rgba(255, 255, 255, 1);

    /**
     * Цвет контента на акцентном фоне с пониженным акцентом.
     */
    --color-accent-content-minor: rgba(255, 255, 255, .3);

    /**
     * Цвет ошибки.
     */
    --color-error: rgba(255, 92, 92, 1);

    /**
     * Цвет ошибки с небольшой прозрачностью. Может быть использован для
     * второстепенных элементов компонентов с ошибками. Например, используется
     * для бордеров полей ввода.
     */
    --color-error-translucent: rgba(255, 92, 92, .85);

    /**
     * Цвет контента для плашек ошибок.
     */
    --color-error-content: rgba(255, 255, 255, .9);
}
