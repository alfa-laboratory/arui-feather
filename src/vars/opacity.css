/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

:root {
    /**
     * Прозрачность по умолчанию.
     */
    --opacity-default: .9;

    /**
     * Прозрачность при наведении или фокусе на элементе.
     */
    --opacity-active: 1;

    /**
     * Прозрачность для второстепенной информации.
     */
    --opacity-minor: .6;

    /**
     * Прозрачность для заблокированных элементов.
     */
    --opacity-disabled: .25;
}
