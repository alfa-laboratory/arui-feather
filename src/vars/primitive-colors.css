:root {
    --color-dark-indigo: #0b1f35;
    --color-dark-indigo-60: rgba(11, 31, 53, 0.6);
    --color-dark-indigo-30: rgba(11, 31, 53, 0.3);
    --color-dark-indigo-15: rgba(11, 31, 53, 0.15);
    --color-dark-indigo-10: rgba(11, 31, 53, 0.1);
    --color-dark-indigo-07: rgba(11, 31, 53, 0.07);
    --color-dark-indigo-05: rgba(11, 31, 53, 0.05);
    --color-dark-indigo-60-flat: #6d7986;
    --color-dark-indigo-30-flat: #b6bcc3;
    --color-dark-indigo-15-flat: #dbdee1;
    --color-dark-indigo-10-flat: #e7e9eb;
    --color-dark-indigo-07-flat: #eeeff1;
    --color-dark-indigo-05-flat: #f3f4f5;
    --color-red-brand: #ef3124;
    --color-red-brand-85: rgba(239, 49, 36, 0.85);
    --color-red-brand-70: rgba(239, 49, 36, 0.7);
    --color-red-brand-50: rgba(239, 49, 36, 0.5);
    --color-red-brand-30: rgba(239, 49, 36, 0.3);
    --color-red-brand-20: rgba(239, 49, 36, 0.2);
    --color-red-brand-10: rgba(239, 49, 36, 0.1);
    --color-red-brand-85-flat: #f15045;
    --color-red-brand-70-flat: #f46f66;
    --color-red-brand-50-flat: #f69791;
    --color-red-brand-30-flat: #fbc2bd;
    --color-red-brand-20-flat: #fcd6d3;
    --color-red-brand-10-flat: #feebea;
    --color-red-brand-05-flat: #fef5f4;
    --color-red-dark: #d91d0b;
    --color-red-error: #ff5c5c;
    --color-green-text: #13a463;
    --color-green-text-60: rgba(19, 164, 99, 0.6);
    --color-green-text-60-flat: #71c8a1;
    --color-green-done: #2fc26e;
    --color-green-done-20: rgba(47, 194, 110, 0.2);
    --color-green-done-10: rgba(47, 194, 110, 0.1);
    --color-green-done-05: rgba(47, 194, 110, 0.05);
    --color-green-done-20-flat: #d5f3e2;
    --color-green-done-10-flat: #ebf9f1;
    --color-green-done-05-flat: #f4fcf7;
    --color-blue-action: #0e60b9;
    --color-blue-swipe-a: #2670c6;
    --color-blue-swipe-b: #3781d8;
    --color-blue-swipe-c: #4a90e2;
    --color-blue-link-mobile: #007aff;
    --color-blue-link-mobile-10: rgba(0, 122, 255, 0.1);
    --color-blue-link-mobile-10-flat: #e6f2ff;
    --color-blue-link-web: #0f62fe;
    --color-sand: #f6bf65;
    --color-lavender: #8888ad;
    --color-amethyst: #a489aa;
    --color-black: #000;
    --color-black-60: rgba(0, 0, 0, 0.6);
    --color-black-30: rgba(0, 0, 0, 0.3);
    --color-black-10: rgba(0, 0, 0, 0.1);
    --color-black-60-flat: #666;
    --color-black-30-flat: #b3b3b3;
    --color-black-10-flat: #e6e6e6;
    --color-white: #fff;
    --color-white-60: rgba(255, 255, 255, 0.6);
    --color-white-30: rgba(255, 255, 255, 0.3);
    --color-white-10: rgba(255, 255, 255, 0.1);
    --color-bg-corporate: #212a33;
    --color-bg-corporate-90: rgba(33, 42, 51, 0.9);
    --color-bg-corporate-60: rgba(33, 42, 51, 0.6);
    --color-bg-corporate-30: rgba(33, 42, 51, 0.3);
    --color-bg-corporate-90-flat: #384048;
    --color-bg-corporate-60-flat: #7a7f85;
    --color-bg-corporate-30-flat: #bdc0c2;
    --color-light-blue: #f5f7f9;
}
