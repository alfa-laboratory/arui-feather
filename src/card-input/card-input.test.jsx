/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

import React from 'react';
import { shallow, mount } from 'enzyme';

import CardInput from './card-input';
import { SCROLL_TO_CORRECTION } from '../vars';

describe('card-input', () => {
    let originalWindowScrollTo = window.scrollTo;

    beforeEach(() => {
        window.scrollTo = jest.fn();
    });

    afterEach(() => {
        window.scrollTo = originalWindowScrollTo;
    });

    it('should render without problems', () => {
        let cardInput = shallow(<CardInput />);

        expect(cardInput).toMatchSnapshot();
    });

    it('should scroll window to element on public `scrollTo` method call', () => {
        let cardInput = mount(<CardInput />);
        let elemTopPosition = cardInput.getDOMNode().getBoundingClientRect().top;
        let elemScrollTo = (elemTopPosition + window.pageYOffset) - SCROLL_TO_CORRECTION;

        cardInput.instance().scrollTo();

        expect(window.scrollTo).toHaveBeenCalledWith(0, elemScrollTo);
    });

    it('should call input focus/blur methods on public focus/blur methods', () => {
        let cardInput = mount(<CardInput />);
        let input = cardInput.instance().root;
        jest.spyOn(input, 'focus');
        jest.spyOn(input, 'blur');


        cardInput.instance().focus();
        expect(input.focus).toHaveBeenCalled();

        cardInput.instance().blur();
        expect(input.blur).toHaveBeenCalled();
    });

    it('should set `type` attribute to `tel`', () => {
        let cardInput = mount(<CardInput />);

        let inputNode = cardInput.find('input');
        expect(inputNode.prop('type')).toBe('tel');
    });
});
