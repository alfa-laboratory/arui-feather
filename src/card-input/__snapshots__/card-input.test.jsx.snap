// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`card-input should render without problems 1`] = `
<Input
  className="card-input card-input_theme_alfa-on-white"
  formNoValidate={true}
  mask="1111 1111 1111 1111 11"
  placeholder="0000 0000 0000 0000 00"
  size="m"
  type="tel"
/>
`;
